js/src/data/selectors.js

```diff
@@ -113,22 +113,28 @@ export const getMCProductStatistics = ( state ) => {
 // which might cause rendering performance problem.
 export const getMCIssues = createSelector(
 	( state, query ) => {
-		if ( ! state.mc_issues ) {
-			return state.mc_issues;
+		const mcIssues = state.mc_issues[ query.issue_type ];
+
+		if ( ! mcIssues ) {
+			return mcIssues;
 		}
 
 		const start = ( query.page - 1 ) * query.per_page;
 		const end = start + query.per_page;
 
 		return {
-			issues: state.mc_issues.issues.slice( start, end ),
-			total: state.mc_issues.total,
+			issues: mcIssues.issues.slice( start, end ),
+			total: mcIssues.total,
 		};
 	},
 	( state ) => [
 		state.mc_issues,
-		state.mc_issues?.issues,
-		state.mc_issues?.total,
+		state.mc_issues.account,
+		state.mc_issues.product,
+		state.mc_issues.account?.issues,
+		state.mc_issues.account?.total,
+		state.mc_issues.product?.issues,
+		state.mc_issues.product?.total,
```

Comment:
💅  To be honest I didn't understand well the purpose of this even before. What was exactly the reason of this? I understood that is for some kind of performance but not sure about the use case.

Reply:
> What was exactly the reason of this? I understood that is for some kind of performance but not sure about the use case.

The reason for using memorizing selectors can be found [here](https://react-redux.js.org/api/hooks#using-memoizing-selectors).

The feature of **rememo** `createSelector` is it not only has the selector dependants but also check if the parameters other than the first one change, to determine whether call the selector to generate the new results.

In May 2021, the **Issues Table** was implemented. To prevent the `issues.slice( start, end )` creating a new reference whenever it's called and further causes the unneeded re-render, we added the `createSelector`. With the addition of the **Status Box** in June, another usage of selector `getMCIssues` was added. Even with `useIsEqualRefValue`, the source parameters of `query` object became two, making the benefit of memorizing somewhat reduced.

With this PR implementation, both issue types use the same selector `getMCIssues` and they are needed for all the same time. It leads the passed-in `query` parameter is also always changing, so the benefits of memorizing are gone. Therefore, I was thinking the **rememo** could be removed completely. Or, split the `query` to `const getMCIssues = createSelector( ( state, issueType, page, perPage ) => {` to make the memorizing work again.

Reply:
And it'd only need the `state.mc_issues` as selector dependants if want to split the `query`.

Reply:
Thanks for the further experiments! 👍 

Sorry I misremembered one thing. I misplaced caching strategy of **rememo** with **memoize-one**. The caching strategy of **rememo** not only caches the result of the most recent calling arguments, it also caches previous results. So the above idea of completely removing **rememo** is wrong, and the benefit of reducing unneeded re-rendering is still working.

However, one thought to clarify is that the purpose of using **rememo** for this selector is to reduce unnecessary re-rendering due to reference changes, not to reduce the number of selector's calling times. This selector isn't an expensive computation. So its number of calling times isn't a major performance consideration when evaluating the effectiveness.

---

📜 
I think it's already ok with the current implementation. If want to enhance it more, changing the selector and resolver of `getMCIssues` by splitting the `query` argument to `issueType`, `page`, `perPage` could reduce the total different amount of results from 13 to 4 in the first-page fetching. And this could be a separate PR.

(The examination method I used is adding a Set in `useMCIssuesTypeFilter` hook )
```js
const results = new Set();
const useMCIssuesTypeFilter = ( issueType = ISSUE_TYPE_ACCOUNT ) => {
	const [ page, setPage ] = useState( defaultQuery.page );

	const { data, hasFinishedResolution } = useAppSelectDispatch(
		'getMCIssues',
		{
			...defaultQuery,
			page,
			issue_type: issueType,
		}
	);
	results.add( data );
	console.log( 'results size', results.size );
	return { data, hasFinishedResolution, page, setPage };
};
```